### ISSUE TITLE
Folder Handlers

### ISSUE THREAD

**Issue by DanielHerr** [WICG/file-handling#9]
Chrome Apps currently have the ability to register as handlers for folders. It would be great to have the same capability as part of this API.

https://developer.chrome.com/apps/manifest/file_handlers

This is what it looks like:
![image](https://user-images.githubusercontent.com/7622517/54098316-1d6cf480-4382-11e9-940f-443003fca9ae.png)

**Comment by fallaciousreasoning** [comment]
Sorry it's taken me so long to get back to you on this.

This is definitely on our roadmap (it'd be really handy for IDE like tools), but probably won't be part of the initial API. We're hoping to add it later, depending on how the initial API is received.

Thanks for taking the time to give feedback :) If you have some specific use cases it'll help us with prioritization.

**Comment by tomayac** [comment]
I think one problem might be OS interoperability. For example, macOS doesn't seem to expose special directory actions. If VS Code is installed, VS Code could offer itself as a directory handler, but I'm not even sure if this is possible on Mac—it's not implemented for the native VS Code app as far as I can tell.

<img width="503" alt="Screenshot 2022-07-25 at 15 50 16" src="https://user-images.githubusercontent.com/145676/180793585-7f03c9ca-dffc-4615-ac4e-49f4ff37b828.png">

Looking at Electron.js (which I believe the native VS Code app is based upon), it seems like passed files are handled as arguments for `process.argv` judging from this [StackOverflow answer](https://stackoverflow.com/questions/70762155/opening-files-with-electron-from-double-click), so the app would then need to open said file. So technically it might be possible to deal with folders here, but I haven't personally seen it in practice.

**Comment by BenjaminAster** [comment]
@tomayac MacOS provides a way to customize the "Quick Actions" menu in the contextmenu of MacOS' finder. It is possible for a human to manually add an "Open folder in VSCode" shortcut to this quick actions menu (see [this StackOverflow answer](https://stackoverflow.com/a/64065309/16523676)). If a human can do this manually, there should probably be a way for browsers to do this automatically.

**Comment by tomayac** [comment]
> Hmm, this is a tricky one indeed. My idea was that it would be somehow possible to control MacOS Automator from the terminal using some fancy commands and add quick actions that way. The browser would then run these terminal commands internally. I could not find much information about this on the internet, except maybe for [this blog post](https://krypted.com/mac-os-x/invoking-automator-workflows-command-line/), which does not seem to be a solution to the problem.

If it involves hackery, I (personally) wouldn't consider it supported.

> On Windows, native applications definitely can register their shortcuts in the "more options" menu that you have already pointed out.

This I would very much consider supported then. So we have at least one major desktop OS with native support. What about Gnome or KDE as major (at least as far as I know) Linux desktop window managers?

**Comment by BenjaminAster** [comment]
> If it involves hackery, I (personally) wouldn't consider it supported.

Yep, the MacOS one is definitely hackery, if it is even possible to register quick actions as an application. But if it is somehow possible, I still think this would be worth it. And if it's not possible, an option would be to provide folder handling only on Windows & Linux.

I've now tried out opening VSCode from the file explorer in both my Zorin OS (uses GNOME) VM...
![Zorin OS folder-right-click context-menu with the option "Open with other application" selected](https://user-images.githubusercontent.com/63414473/180814266-37f77125-b404-4e6b-aff5-be1c09b4d97a.png)
!["Select Application" dialog showing Visual Studio Code as an option](https://user-images.githubusercontent.com/63414473/180814345-72d17c5e-ec2b-48ab-a479-b4fb360cfbd1.png)
...and in my Manjaro (uses KDE Plasma) VM...
![Manjaro folder-right-click context-menu with "Open with" and then "Visual Studio Code" highlighted](https://user-images.githubusercontent.com/63414473/180817927-7998e316-ed7c-4266-84f4-9b7dace036da.png)
...and both desktop environments seem to support opening folders with applications.

**Comment by tomayac** [comment]
Sounds good to me, we seem to have decent enough OS support then. The next step would be to drum up some developer excitement.

**Comment by TheCymaera** [comment]
On Mac, you can open a folder by dragging and dropping it onto the app icon. VSCode supports this.

---

Some applications (such as Godot) register a file associations for the project file, which opens the parent directory for editing. 

Perhaps there should be an option to expose the parent directory handle at launch?

**Comment by alienself** [comment]
Any updates regarding this feature?
